subprojects/core-api/src/main/java/org/gradle/api/artifacts/DependencyMetadata.java

```diff
@@ -43,12 +44,13 @@
      * Returns the version of the module that is targeted by this dependency,
      * which usually expresses what API level of the module you are compatible with.
      */
-    String getVersion();
+    VersionConstraint getVersion();
```

Comment:
:x: Let's call this `getVersionConstraint`

Reply:
I called it `version` because it's called `version {}` for modifying and in the metadata. But.. it's different to `Dependency.getVersion()`. So I guess `getVersionConstraint ()` is what we want. I changed it.